### PR TITLE
[#PAB-443] download endpoint filter programmes and projects on outcom…

### PR DESCRIPTION
**Pull request by cyrusdobbs**
https://trello.com/c/WIx8YKZX/443-download-endpoint-query-update

### Change description
The download query now filters programmes and projects on outcome_category.

- [X] Unit tests and other appropriate tests added or updated
- [X] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### CODE REVIEW

**Comment by cyrusdobbs** [comment]
Need to add areas to the query_params and query

File: core/db/entities.py [hunk 1]
```diff
@@ -289,16 +323,45 @@ def itl_regions(self):
...
+        projects = cls.query.filter(
+            and_(cls.programme_id.in_(programme_ids), cls.submission_id.in_(submission_ids))
+        ).all()
+        return projects
+
+    @classmethod
+    def filter_projects_by_outcome_categories(
+        cls, projects: list["Project"], outcome_categories: list
+    ) -> tuple[list["Project"], list["OutcomeData"]]:
+        """Filter projects by outcome categories.
```
Comment (nsnape) [inline core/db/entities.py:342]: Could you just elaborate a little on the docstring as to what the method is doing, ta

**Review by nsnape — CHANGES_REQUESTED** [review]
Looking pretty good, thanks. Just a couple comments re docstring descriptions, but otherwise would be good to merge this and get it tested :)

File: core/db/entities.py (hunk 1 above)
Comment (cyrusdobbs) [reply core/db/entities.py:342]: Added

File: core/db/entities.py [hunk 2]
```diff
@@ -153,6 +155,38 @@ def get_programmes_by_org_and_fund_type(cls, organisation_ids: list, fund_type_i
...
         return query.all()
 
+    @classmethod
+    def filter_programmes_by_outcome_category(
+        cls, programmes: list, outcome_categories: list
+    ) -> tuple[list["Programme"], list["OutcomeData"]]:
+        """Filter programmes based on outcome categories.
+
+        First filters all programmes to a subset matching the provided ids. If outcome categories are provided, also
+        filters to programmes that have at least on
```
[the remark this answers is not in the document]
Comment (cyrusdobbs) [reply core/db/entities.py:165]: lol... It's also outdated.. changed it now though 👍

**Review by nsnape — APPROVED** [review]

**Pull request merged**